internal/server/resources.go

```
@@ -247,10 +271,18 @@ func (s *Store) updateResource(owner string, capability int, req api.PutResource
 			return "", 0, ErrDropsRoots
 		}
 	}
-	// Replace the GC roots to match the new manifest; chunks only this version
-	// referenced become unreferenced and eligible for a later sweep.
-	if err := replaceResourceChunks(tx, req.ID, owner, req.ChunkRefs); err != nil {
-		return "", 0, err
+	// A refs-less client-GC write leaves any stored rows untouched: they no longer
+	// drive GC on such an account, and deleting them would cost a full-table diff
+	// on every private push for no behavioral difference.
+	if !(req.ClientGC && len(req.ChunkRefs) == 0) {
```

golangci-lint (2.12.2)

[error] QF1001: could apply De Morgan's law

(staticcheck)